cluster-autoscaler/cloudprovider/azure/test/go.mod

```diff
@@ -1,6 +1,8 @@
 module k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/test
 
-go 1.22.3
+go 1.23.0
```

Comment:
This update to go 1.23 checks out. Ref:

- https://cs.opensource.google/go/x/oauth2/+/refs/tags/v0.27.0:go.mod

k/k is at 1.24 in master so I think this is fine

cc @towca @x13n @BigDarkClown

Reply:
nevermind this was in the Azure cloudprovider test package, CAS is already at 1.23

Reply:
> nevermind this was in the Azure cloudprovider test package, CAS is already at 1.23

Yeah, I don't think it's especially important to update the tests, but one of the VPA tests didn't like being out of sync so I just decided to update all of the tests